lua/catppuccin/types.lua

```diff
@@ -3,97 +3,246 @@
+---@class CtpColors<T>: {rosewater: T, flamingo: T, pink: T, mauve: T, red: T, maroon: T, peach: T, yellow: T, green: T, teal: T, sky: T, sapphire: T, blue: T, lavender: T, text: T, subtext1: T, subtext0: T, overlay2: T, overlay1: T, overlay0: T, surface2: T, surface1: T, surface0: T, base: T, mantle: T, crust: T, none: T }
 
 ---@class CatppuccinOptions
----@field background { dark: CtpFlavor, light: CtpFlavor }
----@field compile_path string
+-- Changes the flavor based on the background. See `:h background` for more info.
+---@field background CtpBackground
 -- By default catppuccin writes the compiled results into the system's cache directory.
 -- You can change the cache dir by changing this value.
----@field transparent_background boolean Whether to enable transparency.
----@field show_end_of_buffer boolean Toggle the '~' characters after the end of buffers.
----@field term_colors boolean If true, sets terminal colors (e.g. g:terminal_color_0).
----@field dim_inactive { enabled: boolean, shade: "dark" | "light", percentage: number }
----@field no_italic boolean Disables all italic styles.
----@field no_bold boolean Disables all bold styles.
----@field no_underline boolean Disables all underline styles.
----@field styles CtpStyles Handles the style of general hl groups (see :h highlight-groups).
----@field integrations CtpIntegrations Toggle integrations. Integrations allow Catppuccin to set the theme of various plugins.
----@field color_overrides CtpColors<string> | CtpFlavors<CtpColors<string>> Catppuccin colors can be overwritten here.
----@field highlight_overrides CtpHighlightOverrides Catppuccin highlights can be overwritten here.
----@field custom_highlights CtpHighlightOverrideFn deprecated: use highlight_overrides instead.
----@field flavour CtpFlavor The default flavor to use on startup.
+---@field compile_path string
+-- Whether to enable transparency.
+---@field transparent_background boolean
+-- Toggle the `~` characters after the end of buffers.
+---@field show_end_of_buffer boolean
+-- If true, sets terminal colors (e.g. `g:terminal_color_0`).
+---@field term_colors boolean
+-- Settings for dimming of inactive windows.
+---@field dim_inactive CtpDimInactive
+-- Disables all italic styles.
+---@field no_italic boolean
+-- Disables all bold styles.
+---@field no_bold boolean
+-- Disables all underline styles.
+---@field no_underline boolean
+-- Handles the style of general hl groups (see `:h highlight-groups`).
+---@field styles CtpStyles
+-- Toggle integrations. Integrations allow Catppuccin to set the theme of various plugins.
+---@field integrations CtpIntegrations
+-- Catppuccin colors can be overwritten here.
+---@field color_overrides CtpColors | CtpFlavors<CtpColors<string>>
+-- Catppuccin highlights can be overwritten here.
+---@field highlight_overrides CtpHighlightOverrides
+-- deprecated: use highlight_overrides instead.
+---@field custom_highlights CtpHighlightOverrideFn
```

Comment:
@nullchilly should this be marked as deprecated?
There are still [configs](https://github.com/catppuccin/nvim/blob/7a9437f54ae9d96105dfe41958892e7287adf5ac/tests/setup_spec.lua#L526-L529) using it in our tests, and this annotation alone will not affect the plugin, but might keep people from using it in newer setups.

Reply:
So passing in a table
```lua
custom_highlights = {
  TelescopeTitle = "#ff69b4",
}
```
should also show as correctly typed, gotcha.